docs/running.rst

```diff
@@ -4,115 +4,199 @@
 
-The Push protocol mandates use of SSL/TLS connections to ensure the integrity
-of push subscription data. This rule can be ignored for testing purposes
-only. This requires a valid, trusted TLS certificate (for example, from `Let's
-Encrypt`_).
+``ws://localhost:8080/`` - websocket server
 
-The ``ssl_key`` and ``ssl_cert`` options configure a TLS certificate for
-endpoint's HTTPS and or connection node's WebSocket Secure (WSS) connections:
+``http://localhost:8082/`` - HTTP Endoint Server (See :ref:`the HTTP API <http>`)
 
-.. code-block:: ini
+You could set the ``CRYPTO_KEY`` as an environment variable, or setup a more
+thorough configuration using config files as documented below.
 
-    ; TLS private key and certificate file paths, used for the main
-    ; connection and endpoint listeners. If omitted, TLS will not be
-    ; used.
-    ssl_key = /etc/letsencrypt/live/EXAMPLE.COM/privkey.pem
-    ssl_cert = /etc/letsencrypt/live/EXAMPLE.COM/cert.pem
-    ; Path to file containing any additional parameters for the Ephemeral
-    ; Diffie-Hellman
-    #ssl_dh_param =
+The load-tester can be run against it or you can run Firefox with the
+local Autopush per the :ref:`test-with-firefox` docs.
 
-Certificate options can be easily shared by both node types in
-``autopush_shared.ini`` (e.g. if you have a wildcard certificate). Separate
-credentials can also be specified individually to each node types.
+Configuration
+=============
 
-The connection between the endpoint and connection nodes should not be
-accessible externally but can also have SSL/TLS enabled for added
-protection. This can be configured via the connection node's ``router_ssl_key``
-and ``router_ssl_key`` options:
+Autopush can be configured in three ways; by option flags, by environment variables,
+and by configuration files. Autopush uses three configuration files. These files use
+standard `ini` formatting similar to the following:
 
-.. code-block:: ini
+.. code-block:: cfg
 
-    ; The TLS certificates if this node will be doing SSL termination.
-    router_ssl_key = /etc/letsencrypt/live/EXAMPLE.COM/privkey.pem
-    router_ssl_cert = /etc/letsencrypt/live/EXAMPLE.COM/cert.pem
+   # A comment description
+   ;a_disabled_option
+   ;another_disabled_option=default_value
+   option=value
 
-Running the nodes
-=================
+Options can either have values or act as boolean flags. If the option is a flag
+it is either True if enabled, or False if disabled. The configuration files are
+usually richly commented, and you're encouraged to read them to learn how to
+set up your installation of autopush.
 
-.. code-block:: bash
+Please note: any line that does not begin with a `#` or `;` is considered an option
+line. if an unexpected option is present in a configuration file, the application
+will fail to start.
+
+Configuration files can be located in:
+
+* in the /etc/ directory
+
+* in the configs subdirectory
+
+* in the $HOME or current directory (prefixed by a period '.')
```

Comment:
These are a little ambiguous. What do you think about including the paths alongside each configuration file name?